node/router/src/lib.rs

```diff
@@ -183,7 +185,25 @@ impl<N: Network> Router<N> {
             allow_external_peers,
             storage_mode,
             is_dev,
-        })))
+        }));
+
+        // Ensure there are no bootstrappers among the trusted peers.
+        {
+            let bootstrap_peers = router.bootstrap_peers();
+            let mut peer_pool = router.peer_pool.write();
+            let initial_peer_count = peer_pool.len();
+            peer_pool.retain(|addr, peer| !(peer.is_trusted() && bootstrap_peers.contains(addr)));
```

Comment:
This check should be in start.rs so its for both validators and clients, and so it doesnt include peers stored to disk added to the peer_pool, I think

Reply:
Done in https://github.com/ProvableHQ/snarkOS/pull/3863/commits/967977b2e910babeb95d80d0a632598786ec0663.